util/common/metrics.go

```diff
@@ -130,6 +198,71 @@ func SendCollectDMetrics(metricPerInterval int, sendingInterval, duration time.D
 		}
 	}
 
+}
+func processFile(filePath string, startTime int64) error {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		fmt.Println("Error reading file:", err)
+		return nil
+	}
+
+	//replace START_TIME with the current time
+	modifiedData := strings.ReplaceAll(string(data), "START_TIME", fmt.Sprintf("%d", startTime))
+
+	//curl command
+	url := "http://127.0.0.1:4316/v1/metrics"
+	_, err = http.Post(url, "application/json", bytes.NewBufferString(modifiedData))
+
+	_, err = http.Post(url, "application/json", bytes.NewBufferString(modifiedData))
+	if err != nil {
+		fmt.Println("Failed to send POST request to", url)
+		fmt.Printf("Error: %v\n", err)
+		return err
+	}
+	return nil
+
+}
+
+func SendAppSignalMetrics() error {
+	// The bash script to be executed asynchronously.
+	dir, err := os.Getwd()
+	if err != nil {
+		fmt.Println("Error getting current directory:", err)
+		return err
+	}
+	fmt.Println("Current Directory:", dir)
+
+	// Determine the base directory for the files based on the OS
+	var baseDir string
+	if runtime.GOOS == "windows" {
+		baseDir = "C:\\Users\\Administrator\\amazon-cloudwatch-agent-test\\test\\app_signals\\resources\\metrics"
+	} else { // assuming macOS or Unix-like system
+		baseDir = "/Users/ec2-user/amazon-cloudwatch-agent-test/test/app_signals/resources/metrics"
+	}
+
+	for i := 0; i < 12; i++ {
```

Comment:
why 12? How did we come up with this number?

Reply:
Since we are running for a minute and it sleeps every 5 seconds we will generate 12 times. I'll change it in terms of words to make easier to understand